src/main/java/org/sonarlint/intellij/config/Settings.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2020 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.config;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import org.sonarlint.intellij.config.global.SonarLintGlobalSettings;
+import org.sonarlint.intellij.config.global.SonarLintGlobalSettingsStore;
+import org.sonarlint.intellij.config.module.SonarLintModuleSettings;
+import org.sonarlint.intellij.config.module.SonarLintModuleSettingsStore;
+import org.sonarlint.intellij.config.project.SonarLintProjectSettings;
+import org.sonarlint.intellij.config.project.SonarLintProjectSettingsStore;
+
+import static org.sonarlint.intellij.util.SonarLintUtils.getService;
+
+public class Settings {
```

Comment:
Since this class only exposes static stuff, I think it should have a private "no-op" constructor.

Reply:
I know that this is a common convention among SonarSourcers, it is also raised as a major code smell by next. I usually don't like this convention, for 3 reasons:
* adding a private constructor results in dead code, it is inserted "just in case", so it should not be here. As an exception maybe I can see the point of inserting a private constructor when it's a public API, here it's just internal stuff
* I usually expect people to understand that it's a utility class when all the methods are static. At least personally I don't feel the need to have a private constructor with a comment to explain it
* anyway if you misunderstand and instantiate the class, you can't do anything with it. You basically have a raw object

It could be an opportunity to talk about it on Discuss! 
That said I guess I need to apply the convention anyway

Reply:
This might be an argument in favor of a switch to e.g Kotlin: first class functions, instead of utility class methods.